common/src/math.rs

```diff
@@ -53,79 +89,193 @@ impl<N: Scalar> Deref for MIsometry<N> {
+    /// coordinates, a transformation within the 3D Beltrami-Klein model.
     fn from(value: MIsometry<N>) -> na::Matrix4<N> {
         value.0
     }
 }
 
 impl<N: Scalar> From<MVector<N>> for na::Vector4<N> {
+    /// Unwraps the underlying vector. This effectively reinterprets the vector
+    /// as a vector in Euclidean 4-space, or, if interpreted as homogeneous
+    /// coordinates, a point within the 3D Beltrami-Klein model (as long as it's
+    /// inside the unit ball).
     fn from(value: MVector<N>) -> na::Vector4<N> {
         value.0
     }
 }
 
 impl MIsometry<f32> {
+    /// Casts the components to an `f64`
     pub fn to_f64(self) -> MIsometry<f64> {
         MIsometry(self.0.cast::<f64>())
     }
 }
 
 impl MIsometry<f64> {
+    /// Casts the components to an `f32`
     pub fn to_f32(self) -> MIsometry<f32> {
         MIsometry(self.0.cast::<f32>())
     }
 }
 
 impl MVector<f32> {
+    /// Casts the components to an `f64`
     pub fn to_f64(self) -> MVector<f64> {
         MVector(self.0.cast::<f64>())
     }
 }
 
 impl MVector<f64> {
+    /// Casts the components to an `f32`
     pub fn to_f32(self) -> MVector<f32> {
         MVector(self.0.cast::<f32>())
     }
 }
 
-impl<N: RealField + Copy> MIsometry<N> {
+impl<N: RealField + Copy> AsRef<[[N; 4]; 4]> for MIsometry<N> {
     #[inline]
-    pub fn as_ref(self) -> [[N; 4]; 4] {
-        *self.0.as_ref()
+    fn as_ref(&self) -> &[[N; 4]; 4] {
+        self.0.as_ref()
     }
-    #[inline]
-    pub fn unit_quaternion_to_homogeneous(rotation: na::UnitQuaternion<N>) -> Self {
-        MIsometry(rotation.to_homogeneous())
+}
+
+impl<N: RealField + Copy> From<na::UnitQuaternion<N>> for MIsometry<N> {
+    /// Converts a quaternion into the matrix for the rotation it represents.
+    fn from(value: na::UnitQuaternion<N>) -> Self {
+        MIsometry(value.to_homogeneous())
     }
-    #[inline]
-    pub fn rotation_to_homogeneous(rotation: na::Rotation3<N>) -> Self {
-        MIsometry(rotation.to_homogeneous())
+}
+
+impl<N: RealField + Copy> From<na::Rotation3<N>> for MIsometry<N> {
+    /// Converts a rotation into the matrix representing that rotation.
+    fn from(value: na::Rotation3<N>) -> Self {
+        MIsometry(value.to_homogeneous())
     }
+}
+
+impl<N: RealField + Copy> MIsometry<N> {
+    /// Returns a view containing the i-th row of this matrix.
     #[inline]
-    pub fn row(self, i: usize) -> na::RowVector4<N> {
-        self.0.row(i).into()
+    pub fn row(&self, i: usize) -> na::MatrixView1x4<'_, N, na::U1, na::U4> {
+        self.0.row(i)
     }
+
+    /// Creates an identity matrix.
     #[inline]
     pub fn identity() -> Self {
         Self(na::Matrix4::identity())
     }
-    #[inline]
-    pub fn map<N2: Scalar, F: FnMut(N) -> N2>(&self, f: F) -> MIsometry<N2> {
-        MIsometry(self.0.map(f))
+
+    /// The reflection about the hyperbolic plane represented by the given
+    /// normalized vector. An incorrect matrix will be returned if `a` and `b`
```

Comment:
Note to self: `a` and `b` here don't exist. I copied this part of the comment from `translation` and forgot to update the comment to be accurate.